enclave/core/sgx/init.c

```diff
@@ -60,8 +60,25 @@ static oe_result_t _eeid_patch_memory()
     {
         uint8_t* enclave_base = (uint8_t*)__oe_get_enclave_base();
         uint8_t* heap_base = (uint8_t*)__oe_get_heap_base();
-        oe_eeid_marker_t* marker = (oe_eeid_marker_t*)heap_base;
-        oe_eeid = (oe_eeid_t*)(enclave_base + marker->offset);
+        const oe_eeid_marker_t* marker = (oe_eeid_marker_t*)heap_base;
+        oe_eeid_t* eeid = (oe_eeid_t*)(enclave_base + marker->offset);
+
+        /* EEID must be within the enclave memory */
+        if (!oe_is_within_enclave(eeid, sizeof(oe_eeid_t)) ||
+            !oe_is_within_enclave(
+                eeid->data, eeid->data_size + eeid->signature_size))
+            oe_abort();
+
+        oe_eeid = eeid;
+
+        uint8_t* heap_end = (uint8_t*)__oe_get_heap_end();
+        uint8_t* tcs_end =
+            heap_end + (6 + 2 + eeid->size_settings.num_stack_pages) *
```

Comment:
Please add named definitions for the magic numbers 6 and 2; this source file is well removed from the context of where those numbers come from, e.g.:
```c
#define OE_SGX_TCS_CONTROL_PAGES 6
#define OE_SGX_TCS_GUARD_PAGES 2
```

Reply:
Done!